### PR TITLE
More entry points for serialization compiler plugin

### PR DESCRIPTION
**Pull request by sandwwraith**
* In Kotlin/JS JPS compilation
* In kotlin-common CLI compiler
* In kotlin-common gradle build

### CODE REVIEW

File: compiler/cli/src/org/jetbrains/kotlin/cli/metadata/K2MetadataCompiler.kt [hunk 1]
```diff
@@ -53,6 +54,9 @@ class K2MetadataCompiler : CLICompiler<K2MetadataCompilerArguments>() {
     ): ExitCode {
         val collector = configuration.getNotNull(CLIConfigurationKeys.MESSAGE_COLLECTOR_KEY)
 
+        val plugLoadResult = PluginCliParser.loadPluginsSafe(arguments, configuration)
```
Comment (udalov) [inline compiler/cli/src/org/jetbrains/kotlin/cli/metadata/K2MetadataCompiler.kt]: `val pluginLoadResult` please

File: compiler/cli/src/org/jetbrains/kotlin/cli/metadata/K2MetadataCompiler.kt [hunk 2]
```diff
@@ -53,6 +54,9 @@ class K2MetadataCompiler : CLICompiler<K2MetadataCompilerArguments>() {
     ): ExitCode {
         val collector = configuration.getNotNull(CLIConfigurationKeys.MESSAGE_COLLECTOR_KEY)
 
+        val pluginLoadResult = PluginCliParser.loadPluginsSafe(arguments, configuration)
+        if (pluginLoadResult != ExitCode.OK) return pluginLoadResult
```
Comment (bashor) [inline compiler/cli/src/org/jetbrains/kotlin/cli/metadata/K2MetadataCompiler.kt]: I think would be better to report something here.

cc @udalov

Comment (sandwwraith) [reply compiler/cli/src/org/jetbrains/kotlin/cli/metadata/K2MetadataCompiler.kt]: Some errors reported in `loadPluginsSafe` function:

https://github.com/JetBrains/kotlin/blob/master/compiler/cli/src/org/jetbrains/kotlin/cli/jvm/plugins/PluginCliParser.kt#L43

Comment (udalov) [reply compiler/cli/src/org/jetbrains/kotlin/cli/metadata/K2MetadataCompiler.kt]: Looks fine to me

**Comment by alexey-tsvetkov** [comment]
The changes themselves look good to me

File: compiler/cli/src/org/jetbrains/kotlin/cli/metadata/K2MetadataCompiler.kt (hunk 2 above)
Comment (bashor) [reply compiler/cli/src/org/jetbrains/kotlin/cli/metadata/K2MetadataCompiler.kt]: ok

**Comment by sandwwraith** [comment]
@alexey-tsvetkov commit message itself tells about "entry points for compiler plugins", I've just changed PR title to be more precise. Sorry for misleading.

**Comment by udalov** [comment]
@sandwwraith I suppose this can be closed?

**Pull request closed without merging**